tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -164,17 +162,9 @@ KafkaProducer<byte[], byte[]> createKafkaProducer(Properties props) {
         return new KafkaProducer<>(props);
     }
 
-    static byte[] generateRandomPayload(Integer recordSize, List<byte[]> payloadByteList, byte[] payload,
-            Random random) {
-        if (!payloadByteList.isEmpty()) {
-            payload = payloadByteList.get(random.nextInt(payloadByteList.size()));
-        } else if (recordSize != null) {
-            for (int j = 0; j < payload.length; ++j)
-                payload[j] = (byte) (random.nextInt(26) + 65);
-        } else {
-            throw new IllegalArgumentException("no payload File Path or record Size provided");
-        }
-        return payload;
+    static byte[] select(List<byte[]> payloadByteList,
+                         Random random) {
+        return payloadByteList.get(random.nextInt(payloadByteList.size()));
```

Comment:
I'm thinking, if the record size is 1 or a small number, aren't we still wasting many randonization time for selecting next payload? I'm thinking we random selecting one start index, and then go through all the remaining array 1 by 1. After running back to the starting index, we run random again. WDYT?

Reply:
Yes, I think running through sequentially would be better to prevent repetition. Though I wonder if we need to re-randomise. If each batch can be filled with uniformly distributed data, does it matter if it's re-used?

Maybe it could be conditional? If the user is working with a record size greater than the default message.max.bytes of 1MB use completely randomized data, else use 100MB of pre-generated randomized data. Or make it configurable, a user could configure the amount of random data, or decide between pre-generated and dynamically randomised.